beta-prompts/beta_prompts/improvement_engine.py

```
@@ -0,0 +1,610 @@
+            optimized_prompt=optimized,
+            improvements_applied=improvements_applied,
+            performance_prediction=performance_prediction,
+            analysis=analysis,
+        )
+
+    def _generate_role_definition(self, prompt: str, domain: str) -> str:
+        """Generate appropriate role definition based on prompt content"""
+
+        # Analyze prompt to determine best role
+        if re.search(r"code|program|software|debug|function", prompt, re.IGNORECASE):
+            return f"You are an expert software engineer with deep expertise in {domain} development and best practices."
```

Ruff (0.12.2)

Line too long (121 > 100)

(E501)